src/probnum/typing.py

```diff
@@ -29,6 +29,21 @@
 # Argument Types
 ########################################################################################
 
+try:
+    from numpy.typing import ArrayLike  # pylint: disable=unused-import
+except ImportError:
+    # This is a very basic implementation which may be altered to align more with
+    # NumPy's definition of an array-like
+    _ArrayLikeScalar = Union[np.generic, bool, int, float, complex, str, bytes]
+    _ArrayLikeNestedSequence = Union[  # Dirty hack to avoid recursive types
```

Comment:
Is this sth that needs a better solution in the future (avoiding the "Dirty hack")? Then let's open an Issue about it. I like that you added the comment already mentioning it.

Reply:
I think the best solution is to get rid of it altogether. However, `numpy.typing` only exists since version `1.20`.
By the way, the "dirty hack" is inspired by NumPy's implementation of the thing :D

Reply:
The best solution would be to get rid of it alltogether.
However, `numpy.typing` was introduced in version 1.20.
Fun fact: This implementation is inspired by `numpy`s own implementation to a certain extent 😆